server/interfaces/api/sudoku/sudoku_controller.py

```diff
@@ -1,14 +1,41 @@
+import json
+
 from flask import Blueprint, Response, request
 
-from server.app_sudoku.use_cases.post_validate import post_validate
+from server.app_sudoku.use_cases.post_solve import InvalidSudoku, solve_sudoku
 
 sudoku_v1 = Blueprint("sudoku_v1", __name__)
 
 
-@sudoku_v1.route("/validate", methods=["POST"])
-def validate() -> Response:
+@sudoku_v1.route("/solve", methods=["POST"])
+def solve_endpoint() -> Response:
+    content_type = "application/json"
     data = request.get_json()
 
-    if post_validate(data):
-        return Response(status=200)
-    return Response(status=400)
+    try:
+        sudoku_board = data["sudoku"]
+        solved_sudoku = solve_sudoku(sudoku_board)
+
+        if solved_sudoku:
+            return Response(
+                response=json.dumps({"data": solved_sudoku}),
+                status=200,
+                mimetype=content_type,
+            )
+        return Response(
+            response=json.dumps({"data": "Sudoku is unsolvable"}),
+            status=200,
+            mimetype=content_type,
+        )
+    except KeyError:
+        return Response(
+            response=json.dumps({"error": "Invalid JSON key"}),
+            status=400,
+            mimetype=content_type,
+        )
+    except InvalidSudoku as err:
+        return Response(
+            response=json.dumps({"error": str(err)}),
+            status=400,
+            mimetype=content_type,
+        )
```

Comment:
Som det är nu returneras samma respons "Sudoku is unsolvable" oavsett om det är input av inkorrekt format eller input med inkorrekt format fast olösbar sudoku?

Reply:
"Sudoku is unsolvable" borde bara returneras ifall input är en korrekt sudoku som är olösbar, eller har jag missförstått något längst vägen?

Ska försöka bryta ner hur jag resonerat.

I denna rad `solved_sudoku = solve_sudoku(sudoku_board)` försöker vi lösa sudokun som användaren skickar som input. Om den inte går att lösa får vi en tom `List []` tillbaka och går den att lösa får vi en `List` med en löst sudoku tillbaka.

I nästföljande del av koden, `if solved_sudoku:`, kollas ifall vår `List` är just en tom list, `if solved_sudoku:` resulterar i `False` om `List` är tom. Skulle så vara fallet så returneras `{"data": "Sudoku is unsolvable"}` annar så returneras den lösta sudokun.

Skulle input vara av inkorrekt format så kollas det i denna del av `post_solve.py`:

```python
if not sudoku.is_valid:
    raise InvalidSudoku("Not a valid sudoku")
```

Ett `exception is raised` om input är av inkorrekt format. Detta fångas upp redan när denna rad i `sudoku_controller.py` körs `solved_sudoku = solve_sudoku(sudoku_board)` och vi hoppar då direkt till denna del av koden `except InvalidSudoku as err:` som i sin tur returnerar "Not a valid sudoku".

Reply:
Du har tänkt helt rätt! 

Jag missade raden if `not sudoku.is_valid:` i `post_solve`! Då betyder det förstås att bara om korrekt input är gett så hamnar vi i `solve_sudoku`, om input är felaktigt så får vi ett raiseException och hamnar i except-delen i `solve_endpoint()`. Jag kollade nog med lite trötta ögon. 

Ser jättebra ut, snyggt!